definitions/npm/redux-devtools-extension_v2.x.x/flow_v0.104.x-/redux-devtools-extension_v2.x.x.js

```diff
@@ -1,5 +1,37 @@
-import type { ActionCreator, StoreEnhancer } from 'redux';
-import typeof { compose } from 'redux';
+// Following type definitions are copied from redux_v4 plus inexact type addition to satisfy tests
+// start
+declare type DispatchAPI<A> = (action: A) => A;
```

Comment:
I'm not 100% sure, but I think declaring these types here could potentially cause conflicts (especially if a redux libdef with different definitions for these types is in the same project as this libdef). You may be able to just remove the `declare` keyword from these, but if not adding a prefix or otherwise mangling the name to something unique to this file will work.

Reply:
Good question, I have no experience with that aspect, but I think the concern is totally legit. I don't know if `declare` itself would be the same as `declare export`, from the Flow documentation, the former format alone would mean 'make it global' so that other places within the same project can reference them, but I think they have to be "imported" in order to be used, correct me if I'm wrong. If above statement is true, then there shouldn't be any conflict when we don't import but to declare a new version of a type, there might be a chance that the standalone definition would be out of date as we just copied it over, making it harder to maintain. That's the overhead we need to eat anyway by making this change.

Reply:
I think the primary difference between `declare` and `declare export` is that the latter is used within a `declare module` to declare a ES module's exports. I'm pretty sure the plain `declare` is globally scoped and usable without importing. Though I think it's also possible to use `declare` within `declare module`, at which point what's declared is scoped only to the module.

Reply:
In a libdef I don't think you can have a `type` on its own, and it must be a `declare type`.

That being said, the types being declared must be declared inside the module like Justin had mentioned. Otherwise we run into silent conflicts when consuming libraries that use the same type names. This applies even if the global one would get "shadowed" by all rights by some local import.

Reply:
I think I've tried what Justin suggested last week in Flow REPL, which seems to be fine, but I'm not 100% sure about leaving out the 'declare' keyword  in the actual code base would work without issues, I'll double check with that.

Regarding putting the declare expression in a module, I have a concern on it as this redux-devtools-extension libdef exports multiple modules in a file, which means that I need to 'duplicate' that multiple times in each exported modules?

Reply:
Any types a given module needs will need to be declared in the module directly. That being said, do these submodules need the same types?

Reply:
At a glance, I think they do, as each exported module from that library requires the devToolsEnhancer, which further requires all other dependencies from redux that we borrowed, I don't know if I should refactor the entire file to break those submodules into their own file, we can bring the discussion offline.